### PR TITLE
Updating the Configurable Care application to support FHIR IGs

### PR DESCRIPTION
**Pull request by divyaramnath-13**

### CODE REVIEW

File: configurablecare/app/src/main/java/com/google/fhir/examples/configurablecare/FhirApplication.kt [hunk 1]
```diff
@@ -29,21 +29,31 @@ import com.google.android.fhir.search.search
...
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.launch
+import org.hl7.fhir.utilities.npm.NpmPackage
 import timber.log.Timber
 
 class FhirApplication : Application(), DataCaptureConfig.Provider {
-  private val BASE_URL = "http://10.0.2.2:8080/fhir/"
+  private val BASE_URL = "https://192.168.1.100:8080/fhir/" // "http://127.0.0.1:8080/fhir"  //
```
Comment (jingtang10) [inline configurablecare/app/src/main/java/com/google/fhir/examples/configurablecare/FhirApplication.kt]:
```suggestion
  private val BASE_URL = "https://192.168.1.100:8080/fhir/"
```

File: configurablecare/app/src/main/java/com/google/fhir/examples/configurablecare/AddPatientViewModel.kt [hunk 2]
```diff
@@ -23,29 +23,49 @@ import androidx.lifecycle.viewModelScope
...
+  var currentTargetResourceType: String = ""
+  val context = application.applicationContext
 
   private val questionnaireResource: Questionnaire
     get() =
       FhirContext.forCached(FhirVersionEnum.R4).newJsonParser().parseResource(questionnaire)
         as Questionnaire
   private var fhirEngine: FhirEngine = FhirApplication.fhirEngine(application.applicationContext)
-  private var questionnaireJson: String? = null
+  // private var questionnaireJson: String? = null
```
Comment (jingtang10) [inline configurablecare/app/src/main/java/com/google/fhir/examples/configurablecare/AddPatientViewModel.kt]:
```suggestion
```

File: configurablecare/app/src/main/java/com/google/fhir/examples/configurablecare/AddPatientViewModel.kt [hunk 3]
```diff
@@ -23,29 +23,49 @@ import androidx.lifecycle.viewModelScope
...
 
 /** ViewModel for patient registration screen {@link AddPatientFragment}. */
 class AddPatientViewModel(application: Application, private val state: SavedStateHandle) :
   AndroidViewModel(application) {
 
-  val questionnaire: String
-    get() = getQuestionnaireJson()
-  val savedPatient = MutableLiveData<Patient?>()
+  var questionnaire: String = ""
+  // get() = getQuestionnaireJson()
```
Comment (jingtang10) [inline configurablecare/app/src/main/java/com/google/fhir/examples/configurablecare/AddPatientViewModel.kt]:
```suggestion
```

File: configurablecare/app/src/main/java/com/google/fhir/examples/configurablecare/care/CareWorkflowExecutionViewModel.kt [hunk 4]
```diff
@@ -67,11 +82,19 @@ class CareWorkflowExecutionViewModel(application: Application) : AndroidViewMode
          * exhaustion.
          */
         runBlocking {
-          carePlanManager.applyPlanDefinitionOnPatient(
-            currentPlanDefinitionId,
-            careWorkflowExecutionRequest.patient,
-            getActiveRequestResourceConfiguration()
-          )
+          // carePlanManager.smartIgTest()
```
Comment (jingtang10) [inline configurablecare/app/src/main/java/com/google/fhir/examples/configurablecare/care/CareWorkflowExecutionViewModel.kt]:
```suggestion
```

File: configurablecare/app/src/main/java/com/google/fhir/examples/configurablecare/data/TimestampBasedDownloadWorkManagerImpl.kt [hunk 5]
```diff
@@ -163,6 +163,8 @@ class TimestampBasedDownloadWorkManagerImpl(
       is PlanDefinition -> return extractPlanDefinitionDependentResources(resource)
       is CarePlan -> return extractCarePlanDependentResources(resource)
       is Encounter -> return addEncounterRelatedResources(resource)
+    // is Library -> return carePlanManager.installKnowledgeResource(resource)
+    // is ActivityDefinition -> return carePlanManager.installKnowledgeResource(resource)
```
Comment (jingtang10) [inline configurablecare/app/src/main/java/com/google/fhir/examples/configurablecare/data/TimestampBasedDownloadWorkManagerImpl.kt]:
```suggestion
```

File: configurablecare/app/src/main/java/com/google/fhir/examples/configurablecare/screening/ListActivitiesViewModel.kt [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+                  (orders.size == 0 && plans.size == 0))
+            ) {
+              proposals.add(request)
+            }
+          }
+        }
+        requests = orders + proposals + plans + miscRequests
+      }
+      liveSearchedTasks.value =
+        requests // requestManager.getAllRequestsForPatient(patientId) //, taskStatus)
```
Comment (jingtang10) [inline configurablecare/app/src/main/java/com/google/fhir/examples/configurablecare/screening/ListActivitiesViewModel.kt]:
```suggestion
        requests
```

File: configurablecare/app/src/main/java/com/google/fhir/examples/configurablecare/util/TransformSupportServicesMatchBox.java [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+        return resource;
+//    if (resource != null) {
+//      String inStr = FhirContext.forR4Cached().newJsonParser().encodeResourceToString(resource);
+//      try {
+//        return Manager.parseSingle(context, new ByteArrayInputStream(inStr.getBytes()), FhirFormat.JSON);
+//      } catch (IOException e) {
+//        throw new FHIRException("Cannot convert resource to element model");
+//      }
+//    }
+//    throw new FHIRException("resolveReference, url not found: " + url);
```
Comment (jingtang10) [inline configurablecare/app/src/main/java/com/google/fhir/examples/configurablecare/util/TransformSupportServicesMatchBox.java]:
```suggestion
```

File: configurablecare/app/src/main/java/com/google/fhir/examples/configurablecare/util/ConceptMapEngine.java [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+        }
+        if (tt == null)
+            return null;
+        CanonicalPair cp = new CanonicalPair(cg.getTarget());
+        return new Coding().setSystem(cp.getUrl()).setVersion(cp.getVersion()).setCode(tt.getCode()).setDisplay(tt.getDisplay());
+    }
+
+//    private boolean isOkRelationship(ConceptMapRelationship relationship) {
+//        return relationship != null && relationship != ConceptMapRelationship.NOTRELATEDTO;
+//    }
```
Comment (jingtang10) [inline configurablecare/app/src/main/java/com/google/fhir/examples/configurablecare/util/ConceptMapEngine.java]:
```suggestion
```

File: configurablecare/app/src/main/java/com/google/fhir/examples/configurablecare/util/CustomResourceMapper.kt [hunk 9]
```diff
@@ -0,0 +1,742 @@
...
+
+// import com.google.android.fhir.datacapture.mapping.StructureMapExtractionContext
+// import com.google.android.fhir.datacapture.extensions.initialExpression
+// import com.google.android.fhir.datacapture.extensions.logicalId
+// import com.google.android.fhir.datacapture.extensions.targetStructureMap
+// import com.google.android.fhir.datacapture.extensions.toCodeType
+// import com.google.android.fhir.datacapture.extensions.toCoding
+// import com.google.android.fhir.datacapture.extensions.toIdType
+// import com.google.android.fhir.datacapture.extensions.toUriType
+// import com.google.android.fhir.datacapture.fhirpath.fhirPathEngine
```
Comment (jingtang10) [inline configurablecare/app/src/main/java/com/google/fhir/examples/configurablecare/util/CustomResourceMapper.kt]:
```suggestion
```

**Review by jingtang10 — APPROVED** [review]

**Comment by MJ1998** [comment]
Can we have some description here?
Thanks.

@divyaramnath-13 @jingtang10

**Pull request merged**